model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/client/entity/JpaClientEntity.java

```diff
@@ -583,7 +586,6 @@ public void removeAttribute(String name) {
             JpaClientAttributeEntity attr = iterator.next();
             if (Objects.equals(attr.getName(), name)) {
                 iterator.remove();
-                attr.setClient(null);
```

Comment:
Could you please explain why this change is necessary? AFAIUndersood it is required for syncing when using bidirectonal `@OneToMany` where `@ManyToOne` side manages the association.

Reply:
I found that the `@PreDelete` handler needed to know the parent entity to enforce the locking. When setting it to null, this wan't possible any more.

Then I digged deeper into the Hibernate things and found that "orphanRemoval" should do the job. Then I tested it, and the attribute were removed correctly. "orphanRemoval" is a JPA 2 feature. This might have been necessary in the past, but my manual tests showed that this is no longer necessary; it is sufficient to remove the item from the collection.

Reply:
I see, thank you for the explanation.